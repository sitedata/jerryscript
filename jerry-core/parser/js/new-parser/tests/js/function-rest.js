/* This test contains rest parameter tests. */

function f(...a) {
}

function g(a = 6, b = 7 + 8, ...c) {
}

function h(...a, b) {
}
