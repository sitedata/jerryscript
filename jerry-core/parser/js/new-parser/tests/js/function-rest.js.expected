
--- Script parsing start ---


--- Function parsing start ---


Final byte code dump:

  Maximum stack depth: 1
  Flags: [small_lit_enc,no_lexical_env]
  Argument range end: 0
  Register range end: 1
  Identifier range end: 1
  Const literal range end: 1
  Literal range end: 1

   0 : CBC_RETURN_WITH_BLOCK

Byte code size: 1 bytes

--- Function parsing end ---


--- Function parsing start ---

  [  1] CBC_PUSH_UNDEFINED
  [  1] CBC_STRICT_EQUAL_RIGHT_LITERAL idx:0->var_ident(a)
  [  0] CBC_BRANCH_IF_FALSE_FORWARD
  [  1] CBC_PUSH_NUMBER_POS_BYTE number:6
  [  0] CBC_ASSIGN_SET_IDENT idx:0->var_ident(a)
  [  1] CBC_PUSH_UNDEFINED
  [  1] CBC_STRICT_EQUAL_RIGHT_LITERAL idx:1->var_ident(b)
  [  0] CBC_BRANCH_IF_FALSE_FORWARD
  [  1] CBC_PUSH_NUMBER_POS_BYTE number:7
  [  1] CBC_ADD_RIGHT_LITERAL idx:2->number(8)
  [  0] CBC_ASSIGN_SET_IDENT idx:1->var_ident(b)

Final byte code dump:

  Maximum stack depth: 4
  Flags: [small_lit_enc,no_lexical_env]
  Argument range end: 2
  Register range end: 3
  Identifier range end: 3
  Const literal range end: 4
  Literal range end: 4

   0 : CBC_PUSH_UNDEFINED
   1 : CBC_STRICT_EQUAL_RIGHT_LITERAL idx:0(arg)->var_ident(a)
   3 : CBC_BRANCH_IF_FALSE_FORWARD offset:6(->9)
   5 : CBC_PUSH_NUMBER_POS_BYTE number:6
   7 : CBC_ASSIGN_SET_IDENT idx:0(arg)->var_ident(a)
   9 : CBC_PUSH_UNDEFINED
  10 : CBC_STRICT_EQUAL_RIGHT_LITERAL idx:1(arg)->var_ident(b)
  12 : CBC_BRANCH_IF_FALSE_FORWARD offset:8(->20)
  14 : CBC_PUSH_NUMBER_POS_BYTE number:7
  16 : CBC_ADD_RIGHT_LITERAL idx:3(lit)->number(8)
  18 : CBC_ASSIGN_SET_IDENT idx:1(arg)->var_ident(b)
  20 : CBC_RETURN_WITH_BLOCK

Byte code size: 21 bytes

--- Function parsing end ---


--- Function parsing start ---


--- Script parsing end ---

Parse error 'Rest parameter must be last formal parameter.' at line: 9 col: 18
